c7n/resources/cloudfront.py

```diff
@@ -181,6 +183,66 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('check-s3-origin')
+class CheckS3Origin(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront, and verify ownership.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cfront-test
+                resource: distribution
+                filters:
+                - type: check-s3-origin
+                  accounts_from:
```

Comment:
accounts_from/accounts should be supported if the user doesn't want to host the accounts file in a separate location